[stable/openvpn] systemd-resolved, ip_forward, nodeport, ubuntu fixes

jonstelly opened the pull request:
Signed-off-by: Jon Stelly <967068+jonstelly@users.noreply.github.com>

<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:

    - Added routing / ip_forward issue and workaround to readme
    - Added Ubuntu systemd-resolved and vpn gui issues to readme
    - Move openvpn.conf to openvpn.serverConf for custom lines written to the server config file
    - Add openvpn.clientConf for custom lines written to the client config file
    - Major version bump for the openvpn.conf breaking change
    - If chart is deployed using NodePort service, have clients connect to NodePort instead of externalPort

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

@jasongwartz and @dippynark, let me know if anything needs to be changed.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ x] Chart Version bumped
- [ x] Variables are documented in the README.md
- [x ] Title of the PR starts with chart name (e.g. `[stable/chart]`)

kamsz committed:
ipv4 forwarding init container

Signed-off-by: Kamil Szczygieł <kamil@szczygiel.io>

kamsz committed:
typo

Signed-off-by: Kamil Szczygieł <kamil@szczygiel.io>

jonstelly committed:
openvpn: systemd-resolved, ip_forward, ubuntu fixes

    - routing / ip_forward issue and workaround
    - Ubuntu systemd-resolved and vpn gui issues
    - Move openvpn.conf to openvpn.serverConf for custom lines written to the server config file
    - Add openvpn.clientConf for custom lines written to the client config file
    - Major version bump for these breaking change
    - If chart is deployed using NodePort service, have clients connect to NodePort instead of externalPort

Signed-off-by: Jon Stelly <967068+jonstelly@users.noreply.github.com>

jasongwartz commented on stable/openvpn/README.md:
```diff
@@ -146,3 +147,43 @@ And optionally (see openvpn.taKey setting):
+
+### 1. Routing / ip_forward
+
+Issue: https://github.com/helm/charts/issues/6398
```
Would this be addressed by the following PR? https://github.com/helm/charts/pull/14986

I haven't replicated the issue so I've had trouble reviewing that PR, so I'd be happy to get some help understanding the problem.

jonstelly replied:
I'll test that out today.  If it works I definitely like that solution since the short-running init container has to run privileged but openvpn can then be run normally.  If that PR fixes the ip_forwarding I'll remove this bit from the readme

For replicating the issue, it definitely shows up in clusters deployed with RKE.  As I understand it it's something to do with the way that RKE spins up kubelet in a container.

jonstelly replied:
The init container works for me on an RKE deployed cluster.  Updated the readme to suggest setting that value instead of the workaround

jasongwartz replied:
cool, thanks for checking! then we'd just have to make sure that PR gets merged before this one.

jasongwartz commented on stable/openvpn/values.yaml:
```diff
@@ -87,10 +87,19 @@ openvpn:
+  # serverConf: |
   #  max-clients 100
   #  client-to-client
+  # Lines appended to the end of the client configuration file
```
The formatting of this is a bit confusing to read - could you add a newline inbetween, or add some kind of clarification to indicate that this comment now applies to the `clientConf` and isn't a continuation of the `serverConf` example?

jonstelly replied:
Added blank line and better word wrapping for comment.

jasongwartz replied:
:ok_hand:

jasongwartz reviewed (commented):
thanks for the contribution! overall looks good, just a couple things to clarify. in the meantime, i'll trigger the tests

jonstelly committed:
Updates per PR comments/suggestions

Signed-off-by: Jon Stelly <967068+jonstelly@users.noreply.github.com>

jonstelly committed:
merge from master

Signed-off-by: Jon Stelly <967068+jonstelly@users.noreply.github.com>

jonstelly committed:
remove whitespace from values.yaml

Signed-off-by: Jon Stelly <967068+jonstelly@users.noreply.github.com>

kamsz committed:
Merge branch 'master' of github.com:helm/charts into ipv4-forward-openvpn

jonstelly committed:
merge updates from master

jonstelly committed:
merge from master

Signed-off-by: Jon Stelly <967068+jonstelly@users.noreply.github.com>

jonstelly committed:
Merge remote-tracking branch 'upstream/master' into openvpn-fixes

jonstelly committed:
Merge remote-tracking branch 'other/ipv4-forward-openvpn' into openvpn-fixes

jonstelly committed:
Add newlines

- Ensure newClientCert.sh contains newlines for each config option, prevents multiple openvpn directives from ending up on same line and being invalid

Signed-off-by: Jon Stelly <967068+jonstelly@users.noreply.github.com>

jonstelly committed:
Merge remote-tracking branch 'upstream/master' into openvpn-fixes
